cmd/state/main.go

```diff
@@ -100,7 +100,10 @@ func run(args []string, isInteractive bool, out output.Outputer) (int, error) {
 	verbose := os.Getenv("VERBOSE") != "" || argsHaveVerbose(args)
 	logging.CurrentHandler().SetVerbose(verbose)
 
-	cfg := config.Get()
+	cfg, err := config.Get()
```

Comment:
We should rename this to `New()`. `Get()` to me conveys the deprecated "global" way of doing things.

Reply:
`Get()` is currently still a "global" function as it returns a persistent configuration instance - if that exists.  It just now also returns an error.
I didn't remove that behavior yet, because

- `analytics` and `locale` use the configuration inside of their `init()` functions
- Changing it in the authentication package, means updating every `authentication.Get()`, `authentication.Client()` and `authentication.ClientAuth()` call, and that happens in 127 (!) places. I forgot to make a story for that though, which I did now: https://www.pivotaltracker.com/story/show/176419803

I think that I will feed the configuration instance into the remaining places first.  Then I will see if I could try to change the `locale` and `analytics` packages, as you suggested... by adding a `SetConfig()` method from the main function.

I could also give the `auth` refactor a try, of course...   But maybe it is better to first refactor the `authentication` package, such that it does not have the globally persisting object anymore.

Reply:
Ok, I guess it makes sense to scope this down into another story.

Reply:
https://www.pivotaltracker.com/story/show/176422561